optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberAccessorImplementor.java

```diff
@@ -0,0 +1,842 @@
+            return (MemberAccessor) gizmoClassLoader.loadClass(className)
+                    .getConstructor().newInstance();
+        } catch (InvocationTargetException | InstantiationException | IllegalAccessException | ClassNotFoundException
+                | NoSuchMethodException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    /**
+     * Returns the Generic Type that a particular
+     * MemberAccessor should return. Used in generated
+     * Gizmo code.
+     *
+     * @param gizmoMemberAccessorName The MemberAccessor that is being queried
+     * @return The generic type gizmoMemberAccessorName should return
+     *         in MemberAccessor.getGenericType
+     */
+    public static Type getGenericTypeFor(String gizmoMemberAccessorName) {
+        return gizmoMemberAccessorNameToGenericType.get(gizmoMemberAccessorName);
+    }
+
+    /**
+     * Returns the AnnotatedElement that a particular
+     * MemberAccessor should return. Used in generated
+     * Gizmo code.
+     *
+     * @param gizmoMemberAccessorName The MemberAccessor that is being queried
+     * @return The AnnotatedElement gizmoMemberAccessorName should return
+     *         in MemberAccessor.getAnnotatedElement
+     */
+    public static AnnotatedElement getAnnotatedElementFor(String gizmoMemberAccessorName) {
+        return gizmoMemberAccessorNameToAnnotatedElement.get(gizmoMemberAccessorName);
+    }
+
+    // ************************************************************************
+    // MemberAccessor methods
+    // ************************************************************************
+
+    private static MethodCreator getMethodCreator(ClassCreator classCreator, String methodName, Class<?>... parameters) {
+        try {
+            return classCreator.getMethodCreator(
+                    MethodDescriptor.ofMethod(MemberAccessor.class.getMethod(methodName, parameters)));
+        } catch (NoSuchMethodException e) {
+            throw new IllegalStateException("No such method: " + methodName, e);
+        }
+    }
+
+    private static void createConstructor(ClassCreator classCreator, MemberDescriptor member,
+            Class<? extends Annotation> annotationClass) {
+        MethodCreator methodCreator =
+                classCreator.getMethodCreator(MethodDescriptor.ofConstructor(classCreator.getClassName()));
+
+        ResultHandle thisObj = methodCreator.getThis();
+
+        // Invoke Object's constructor
+        methodCreator.invokeSpecialMethod(MethodDescriptor.ofConstructor(Object.class), thisObj);
+
+        // Return this (it a constructor)
+        methodCreator.returnValue(thisObj);
+    }
+
+    /**
+     * Generates the following code:
+     *
+     * Class getDeclaringClass() {
+     * return ClassThatDeclaredMember.class;
+     * }
+     */
+    private static void createGetDeclaringClass(ClassCreator classCreator, MemberDescriptor member,
+            Class<? extends Annotation> annotationClass) {
```

Comment:
See above; it is used in one method for an exception message (fails MemberAccessor tests that check for a specific error message otherwise). (So to be consistent, it included in all methods)